### PR TITLE
[IMP] l10n_in: option to enable audit trail

### PR DESCRIPTION
**Pull request by niai-odoo**
Before This Commit:
Currently, the Audit trail feature is applied to all Indian companies.

After This Commit:
The audit trail feature is optional now. it can be enabled in the company.

Task - 3614528

### CODE REVIEW

**Review by jva-odoo — APPROVED** [review]

**Comment by jva-odoo** [comment]
@jco-odoo

File: addons/l10n_in/models/company.py [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
+from odoo.exceptions import UserError
 
 class ResCompany(models.Model):
     _inherit = 'res.company'
 
     l10n_in_upi_id = fields.Char(string="UPI Id")
+    l10n_in_active_audit_trail = fields.Boolean(string="Audit Trail", default=False, help="Once you activate audit trail it can't be deactivated.")
+
+    def write(self, vals):
+        if 'l10n_in_active_audit_trail' in vals and self._origin.l10n_in_active_audit_trail and not vals['l10n_in_active_audit_trail']:
```
Comment (jco-odoo) [inline addons/l10n_in/models/company.py:11]: do we need an ._origin in writes?  Also, self can be multiple records...  It is weird as well that you could be change put the setting and you won't be able to set it back after.  We could just have a tracking True.  @jva-odoo

Comment (jva-odoo) [reply addons/l10n_in/models/company.py]: it's mandatory that once you activate it no one can change it.
image

Comment (ajjo-odoo) [reply addons/l10n_in/models/company.py]: Audit Trail Notification

Comment (ajjo-odoo) [reply addons/l10n_in/models/company.py]: @jco-odoo

Comment (ajjo-odoo) [reply addons/l10n_in/models/company.py]: https://drive.google.com/file/d/1ahIAJZDAG-dh8YlD74DkERDQ06QXQeMm/view?pli=1

Full Notification

Comment (ajjo-odoo) [reply addons/l10n_in/models/company.py]: As this Notification is issued to Ministry of Corporate Affairs ( Governing body of Private, Public & Limited Liability companies, so Applicability of this Audit Trail would only limited to such nature of companies.

where in if organisation is Proprietary, Partnership, Hindu Undivided Family, Trust, NGO, Association of Person than this Audit Trail notification will not apply.

Comment (jco-odoo) [reply addons/l10n_in/models/company.py]: @tsb-odoo It makes more sense with this explanation ^

File: addons/l10n_in/models/company.py [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
 class ResCompany(models.Model):
     _inherit = 'res.company'
 
     l10n_in_upi_id = fields.Char(string="UPI Id")
+    l10n_in_active_audit_trail = fields.Boolean(string="Audit Trail", default=False, help="Once you activate audit trail it can't be deactivated.")
+
+    def write(self, vals):
+        if 'l10n_in_active_audit_trail' in vals:
+            for company in self:
+                if company._origin.l10n_in_active_audit_trail and not vals['l10n_in_active_audit_trail']:
```
[the remark this answers is not in the document]
Comment (niai-odoo) [reply addons/l10n_in/models/company.py]: Done

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**